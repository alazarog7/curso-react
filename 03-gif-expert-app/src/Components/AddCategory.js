import React, { useState } from 'react'
import PropType from 'prop-types';
const AddCategory = ({setcategory})=> {
    
    const [inputValue, setinputValue] = useState("");

    const handleInputChange = (e)=>{
       setinputValue(e.target.value);
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        if(inputValue.length > 0){
            setcategory([inputValue])
            setinputValue('');
        }
        
    } 

    return (
        <form onSubmit={handleSubmit}>
           <h1>{inputValue}</h1>
           <input 
                type="text" 
                value={inputValue}
                onChange = {handleInputChange} 
           />
        </form>
    )
}

AddCategory.propTypes = {
    setcategory: PropType.func.isRequired 
}


export default AddCategory;