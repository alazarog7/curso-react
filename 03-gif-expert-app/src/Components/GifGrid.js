import React, { useState, useEffect } from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import { PropTypes } from 'prop-types';

const GifGrid = ({category}) => {

    //const [images, setImages] = useState([]);
    const { data:images, loading} = useFetchGifs(category);
   
    return (
        <>
            <h3 className="animate__zoomInLeft">{category}</h3>
            {loading && <p>Cargando ...</p> }
            <div className="card-grid">
                {
                    images.map( img => <GifGridItem key={img.id} {...img}/> )
                }
            </div>
        </>
    );
};

export default GifGrid;


GifGrid.propTypes={
    category: PropTypes.string.isRequired
}

