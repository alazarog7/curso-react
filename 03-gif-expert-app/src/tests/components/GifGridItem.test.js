import React from 'react';
c
import { GifGridItem } from "../../Components/GifGridItem";

describe('Prueba del componente GifGridItem', () => {
    
    const title = 'Titulo de prueba';
    const link = 'http://localhost/imagen.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={link}/>);


    test('should print the template', () => {
        expect(wrapper).toMatchSnapshot();
    }); 

    test('should a has a paragraph with the title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim()).toBe(title);
    })

    test('should has the url and alt same as props', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(link);
        expect(img.props().alt).toBe(title);
    })

    test('should has de animated__fadeIn class', () => {
        const div = wrapper.find('div');
        expect(div.props().className.includes("bounce")).toBe(true);
    })


    
    
    

})
