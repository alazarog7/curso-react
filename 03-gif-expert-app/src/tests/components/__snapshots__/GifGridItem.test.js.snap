// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente GifGridItem should print the template 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="Titulo de prueba"
    height="100px"
    src="http://localhost/imagen.jpg"
    width="100px"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
